controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -121,13 +123,22 @@ type Workload struct {
 var _ WorkloadCluster = &Workload{}
 
 func (w *Workload) getControlPlaneNodes(ctx context.Context) (*corev1.NodeList, error) {
-	nodes := &corev1.NodeList{}
-	labels := map[string]string{
-		labelNodeRoleControlPlane: "",
-	}
-	if err := w.Client.List(ctx, nodes, ctrlclient.MatchingLabels(labels)); err != nil {
+	allNodes := &corev1.NodeList{}
+	if err := w.Client.List(ctx, allNodes); err != nil {
 		return nil, err
 	}
+
+	nodes := &corev1.NodeList{}
+	for i := range allNodes.Items {
+		node := allNodes.Items[i]
+
```

Comment:
can we have unit tested `isControlPlaneNode()` which does the annotations logic?

Reply:
Done